service/matching/task_queue_partition_manager.go

```diff
@@ -96,51 +105,98 @@ func newTaskQueuePartitionManager(
 		tqConfig,
 		partition.TaskQueue().TaskType())
 	pm := &taskQueuePartitionManagerImpl{
-		engine:           e,
-		partition:        partition,
-		ns:               ns,
-		config:           tqConfig,
-		logger:           logger,
-		throttledLogger:  throttledLogger,
-		matchingClient:   e.matchingRawClient,
-		metricsHandler:   metricsHandler,
-		versionedQueues:  make(map[PhysicalTaskQueueVersion]physicalTaskQueueManager),
-		userDataManager:  userDataManager,
-		rateLimitManager: rateLimitManager,
-	}
+		engine:                e,
+		partition:             partition,
+		ns:                    ns,
+		config:                tqConfig,
+		logger:                logger,
+		throttledLogger:       throttledLogger,
+		matchingClient:        e.matchingRawClient,
+		metricsHandler:        metricsHandler,
+		versionedQueues:       make(map[PhysicalTaskQueueVersion]physicalTaskQueueManager),
+		userDataManager:       userDataManager,
+		rateLimitManager:      rateLimitManager,
+		defaultQueueFuture:    future.NewFuture[physicalTaskQueueManager](),
+		autoEnableRateLimiter: quotas.NewRateLimiter(1.0/60, 1),
+	}
+	pm.initCtx, pm.initCancel = context.WithCancel(context.Background())
 
 	if pm.partition.IsRoot() {
 		pm.cache = cache.New(10000, &cache.Options{
 			TTL: max(1, tqConfig.TaskQueueInfoByBuildIdTTL())}, // ensure TTL is never zero (which would disable TTL)
 		)
 	}
 
+	return pm, nil
+}
+
+func (pm *taskQueuePartitionManagerImpl) initialize() (retErr error) {
+	defer func() { pm.defaultQueueFuture.SetIfNotReady(nil, retErr) }()
 	unload := func(bool) {
+		pm.logger.Debug("unloading partitionManager due to config change")
 		pm.unloadFromEngine(unloadCauseConfigChange)
 	}
 
-	var fairness bool
-	fairness, pm.cancelFairnessSub = tqConfig.EnableFairnessSub(unload)
-	// Fairness is disabled for sticky queues for now so that we can still use TTLs.
-	tqConfig.EnableFairness = fairness && partition.Kind() != enumspb.TASK_QUEUE_KIND_STICKY
-	if fairness {
-		tqConfig.NewMatcher = true
-	} else {
-		tqConfig.NewMatcher, pm.cancelNewMatcherSub = tqConfig.NewMatcherSub(unload)
+	retErr = pm.userDataManager.WaitUntilInitialized(pm.initCtx)
```

Comment:
nitpick, but what I meant was to name retErr just so that the defer could see it, but otherwise don't change the code, i.e. still `return err` or `return serviceerror.NewInternal...`

then you don't have any confusion about whether it's being shadowed or not. I think it's correct as-is, just not the usual style.